contracts/contracts/strategies/sonic/SonicSwapXAMOStrategy.sol

```diff
@@ -0,0 +1,738 @@
+                IPair(_baseConfig.platformAddress).token1() == _os,
+            "Incorrect pool tokens"
+        );
+        // Checked both tokens are to 18 decimals
+        require(
+            IBasicToken(_ws).decimals() == 18 &&
+                IBasicToken(_os).decimals() == 18,
+            "Incorrect token decimals"
+        );
+        // Check the SwapX pool is a Stable AMM (sAMM)
+        require(
+            IPair(_baseConfig.platformAddress).isStable() == true,
+            "Pool not stable"
+        );
+        // Check the gauge is for the pool
+        require(
+            IGauge(_gauge).TOKEN() == _baseConfig.platformAddress,
+            "Incorrect gauge"
+        );
+
+        // Set the immutable variables
+        os = _os;
+        ws = _ws;
+        pool = _baseConfig.platformAddress;
+        gauge = _gauge;
+
+        // This is an implementation contract. The governor is set in the proxy contract.
+        _setGovernor(address(0));
+    }
+
+    /**
+     * Initializer for setting up strategy internal state. This overrides the
+     * InitializableAbstractStrategy initializer as SwapX strategies don't fit
+     * well within that abstraction.
+     * @param _rewardTokenAddresses Array containing SWPx token address
+     */
+    function initialize(address[] calldata _rewardTokenAddresses)
+        external
+        onlyGovernor
+        initializer
+    {
+        address[] memory pTokens = new address[](1);
+        pTokens[0] = pool;
+
+        address[] memory _assets = new address[](1);
+        _assets[0] = ws;
+
+        InitializableAbstractStrategy._initialize(
+            _rewardTokenAddresses,
+            _assets,
+            pTokens
+        );
+
+        _approveBase();
+    }
+
+    /***************************************
+                    Deposit
+    ****************************************/
+
+    /**
+     * @notice Deposit an amount of Wrapped S (wS) into the SwapX pool.
+     * Mint OS in proportion to the pool's wS and OS reserves,
+     * transfer Wrapped S (wS) and OS to the pool,
+     * mint the pool's LP token and deposit in the gauge.
+     * @dev This tx must be wrapped by the VaultValueChecker.
+     * To minimize loses, the pool should be rebalanced before depositing.
+     * @param _asset Address of Wrapped S (wS) token.
+     * @param _wsAmount Amount of Wrapped S (wS) tokens to deposit.
+     */
+    function deposit(address _asset, uint256 _wsAmount)
+        external
+        override
+        onlyVault
+        nonReentrant
+        skimPool
+    {
+        require(_asset == ws, "Unsupported asset");
+        require(_wsAmount > 0, "Must deposit something");
+
+        (uint256 osDepositAmount, ) = _deposit(_wsAmount);
+
+        // Emit event for the deposited wS tokens
+        emit Deposit(ws, pool, _wsAmount);
+        // Emit event for the minted OS tokens
+        emit Deposit(os, pool, osDepositAmount);
```

Comment:
The deposit and withdraw events are from the vaults perspective. What user funds is the vault investing and withdrawing. We don't emit events in our AMO's for minted protocol owned liquidity.

(Applies to all deposit/withdraw events in this contract)

Reply:
The Curve AMO emits Deposit and Withdraw events for protocol owned liquidity.

Here's the `Deposit` event for OETH in the OETH Curve AMO https://github.com/OriginProtocol/origin-dollar/blob/b4544dce7751298067b18084ac326142d82aef13/contracts/contracts/strategies/ConvexEthMetaStrategy.sol#L212

The `Deposit` event is also emitted in the `mintAndAddOTokens` rebalance function https://github.com/OriginProtocol/origin-dollar/blob/b4544dce7751298067b18084ac326142d82aef13/contracts/contracts/strategies/ConvexEthMetaStrategy.sol#L414